common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+        if ( gpio_cfg[g_idx * 32 + index].is_init == DISABLE ) {
+            shell_print(shell, "[%-3d][%s %-3d] %-35s: -- | %-9s | NA",
+                g_idx*32+index, dev->name, index, "gpio_disable", "i/o");
+            continue;
+        }
+
+#if PINMASK_RESERVE_CHECK
+        /* avoid pin_mask from devicetree "gpio-reserved" */
+        if (gpio_check_reserve(dev, index, CHECK_BY_GROUP_IDX)) {
+            shell_print(shell, "[%-3d][%s %-3d] %-35s: -- | %-9s | NA",
+                g_idx*32+index, dev->name, index, "gpio_reserve", "i/o");
+            continue;
+        }
+#endif
+        char *pin_dir = "output";
+        if ( gpio_cfg[g_idx * 32 + index].direction == GPIO_INPUT )
+            pin_dir = "input";
+
+        char *pin_dir_reg = "I";
+        if ( g_dir & BIT(index) )
+            pin_dir_reg = "O";
+
+        char *pin_prop = (gpio_cfg[g_idx * 32 + index].property == OPEN_DRAIN) ? "OD":"PP";
+
+        rc = gpio_pin_get(dev, index);
+        if (rc >= 0) {
+            shell_print(shell, "[%-3d][%s %-3d] %-35s: %2s | %-6s(%s) | %d(%d)",
+                g_idx*32+index, dev->name, index, gpio_get_name(dev->name, index),
+                pin_prop, pin_dir, pin_dir_reg, rc, GET_BIT_VAL(g_val, index) );
+        } else {
+            shell_error(shell, "[%-3d][%s %-3d] %-35s: %2s | %-6s | err[%d]",
+                g_idx*32+index, dev->name, index, gpio_get_name(dev->name, index),
+                pin_prop, pin_dir, rc);
+        }
+    }
+
+    return;
+}
+
+static void cmd_gpio_cfg_list_all(const struct shell *shell, size_t argc, char **argv)
+{
+    if (argc != 1) {
+        shell_warn(shell, "Try: platform gpio list_all");
+        return;
+    }
+
+    for (int gpio_idx=0; gpio_idx<GPIO_CFG_SIZE; gpio_idx++)
+        gpio_access_cfg(shell, gpio_idx, GPIO_READ, NULL);
+
+    return;
+}
+
+static void cmd_gpio_cfg_get(const struct shell *shell, size_t argc, char **argv)
+{
+    if (argc != 2) {
+        shell_warn(shell, "Try: platform gpio get <gpio_idx>");
+        return;
+    }
+
+    int gpio_index = strtol(argv[1], NULL, 10);
+    if ( gpio_access_cfg(shell, gpio_index, GPIO_READ, NULL) )
+        shell_error(shell, "gpio[%d] get failed!", gpio_index);
+
+    return;
+}
+
+static void cmd_gpio_cfg_set_val(const struct shell *shell, size_t argc, char **argv)
+{
+    if (argc != 3) {
+        shell_warn(shell, "Try: platform gpio set val <gpio_idx> <data>");
+        return;
+    }
+
+    int gpio_index = strtol(argv[1], NULL, 10);
+    int data = strtol(argv[2], NULL, 10);
+
+    if ( gpio_access_cfg(shell, gpio_index, GPIO_WRITE, &data) )
+        shell_error(shell, "gpio[%d] --> %d ,failed!", gpio_index, data);
+    else
+        shell_print(shell, "gpio[%d] --> %d ,success!", gpio_index, data);
+    
+    return;
+}
+
+static void cmd_gpio_cfg_set_dir(const struct shell *shell, size_t argc, char **argv)
+{
+    shell_warn(shell, "GPIO set DIR command not support!");
```

Comment:
...not supported!"

Reply:
Got it, this suggestion would add in next pr, thanks!